docs/how-to/service-dependencies.md

```diff
@@ -1,11 +1,297 @@
+
+```{terminal}
+   :input: pebble run
+2024-06-28T02:28:06.569Z [pebble] Started daemon.
+[database backend-server frontend-server]
+2024-06-28T02:28:06.575Z [pebble] POST /v1/services 3.212375ms 202
+2024-06-28T02:28:06.578Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T02:28:06.627Z [pebble] Change 1 task (Start service "database") failed: cannot start service: exited quickly with code 1
+2024-06-28T02:28:06.633Z [pebble] GET /v1/changes/1/wait 57.610375ms 200
+2024-06-28T02:28:06.633Z [pebble] Started default services with change 1.
+```
+
+If we check the state of the services:
+
+```{terminal}
+   :input: pebble services
+Service          Startup  Current   Since
+backend-server   enabled  inactive  -
+database         enabled  inactive  -
+frontend-server  enabled  inactive  -
+```
+
+We can see that since the database failed to start, the backend and the frontend are not started. We can further verify this in Pebble [changes and tasks](../reference/changes-and-tasks/changes-and-tasks.md):
+
+Change:
+
+```{terminal}
+   :input: pebble changes
+ID   Status  Spawn               Ready               Summary
+1    Error   today at 10:28 CST  today at 10:28 CST  Autostart service "database" and 2 more
+```
+
+Task:
+
+```{terminal}
+   :input: pebble tasks 1
+Status  Spawn               Ready               Summary
+Error   today at 10:28 CST  today at 10:28 CST  Start service "database"
+Hold    today at 10:28 CST  today at 10:28 CST  Start service "backend-server"
+Hold    today at 10:28 CST  today at 10:28 CST  Start service "frontend-server"
+
+......................................................................
+Start service "database"
+
+2024-06-28T10:28:06+08:00 INFO Most recent service output:
+    Traceback (most recent call last):
+      File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
+        return _run_code(code, main_globals, None,
+      File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
+        exec(code, run_globals)
+      File "/usr/lib/python3.10/http/server.py", line 1307, in <module>
+        test(
+      File "/usr/lib/python3.10/http/server.py", line 1258, in test
+        with ServerClass(addr, HandlerClass) as httpd:
+      File "/usr/lib/python3.10/socketserver.py", line 452, in __init__
+        self.server_bind()
+      File "/usr/lib/python3.10/http/server.py", line 1301, in server_bind
+        return super().server_bind()
+      File "/usr/lib/python3.10/http/server.py", line 137, in server_bind
+        socketserver.TCPServer.server_bind(self)
+      File "/usr/lib/python3.10/socketserver.py", line 466, in server_bind
+        self.socket.bind(self.server_address)
+    OSError: [Errno 98] Address already in use
+2024-06-28T10:28:06+08:00 ERROR cannot start service: exited quickly with code 1
+```
+
+We can see that since the database failed to start, other services that depend on it are put in "Hold" status.
```

Comment:
"Hold" is actually a changes & task thing, not a service status. Maybe this:

```suggestion
We can see that since the database failed to start, other services that depend on it are not started (their service status remains "inactive").
```